liquidctl/driver/aquacomputer.py

```diff
@@ -340,17 +354,79 @@ def set_speed_profile(self, channel, profile, **kwargs):
         elif self._device_info["type"] == self._DEVICE_FARBWERK360:
             raise NotSupportedByDevice()
 
-    def set_fixed_speed(self, channel, duty, **kwargs):
+    def _set_fixed_speed_hwmon(self, channel, duty):
+        hwmon_pwm_name = self._device_info["hwmon_ctrl_mapping"][channel]
+        hwmon_pwm_enable_name = f"{hwmon_pwm_name}_enable"
+
+        if self._hwmon.has_attribute(hwmon_pwm_name) and self._hwmon.has_attribute(
+            hwmon_pwm_enable_name
+        ):
+            # Set channel to direct percent mode
+            self._hwmon.write_str(hwmon_pwm_enable_name, "1")
+
+            # Convert duty from percent to PWM range (0-255)
+            pwm_duty = duty * 255 // 100
+
+            # Write to hwmon
+            self._hwmon.write_str(hwmon_pwm_name, str(pwm_duty))
+        else:
+            _LOGGER.error(
+                "cannot write to %s driver, required PWM functionality is not available",
+                self._hwmon.driver,
+            )
+
+    def _set_fixed_speed_directly(self, channel, duty):
+        # Request an up to date ctrl report
+        report_length = self._device_info["ctrl_report_length"]
+        ctrl_settings = self.device.get_feature_report(_AQC_CTRL_REPORT_ID, report_length)
+
+        fan_ctrl_offset = self._device_info["fan_ctrl"][channel]
+
+        # Set fan to direct percent-value mode
+        ctrl_settings[fan_ctrl_offset + _AQC_FAN_TYPE_OFFSET] = 0
+
+        # Write down duty for channel
+        put_unaligned_be16(
+            duty * 100,  # Centi-percent
+            ctrl_settings,
+            fan_ctrl_offset + _AQC_FAN_PERCENT_OFFSET,
+        )
+
+        # Update checksum value at the end of the report
+        crc16usb_func = crcmod.predefined.mkCrcFun("crc-16-usb")
+
+        checksum_part = bytes(ctrl_settings[0x01 : report_length - 3 + 1])
+        checksum_bytes = crc16usb_func(checksum_part)
+        put_unaligned_be16(checksum_bytes, ctrl_settings, report_length - 2)
+
+        self.device.send_feature_report(ctrl_settings)
+
+    def set_fixed_speed(self, channel, duty, direct_access=False, **kwargs):
         if (
-            self._device_info["type"] == self._DEVICE_D5NEXT
-            or self._device_info["type"] == self._DEVICE_OCTO
+            self._device_info["type"] == self._DEVICE_OCTO
             or self._device_info["type"] == self._DEVICE_QUADRO
         ):
             # Not yet implemented
             raise NotSupportedByDriver()
         elif self._device_info["type"] == self._DEVICE_FARBWERK360:
             raise NotSupportedByDevice()
 
+        # Clamp duty between 0 and 100
+        duty = clamp(duty, 0, 100)
+
+        if self._hwmon and not direct_access:
+            _LOGGER.info(
+                "bound to %s kernel driver, writing fixed speed to hwmon", self._hwmon.driver
+            )
+            return self._set_fixed_speed_hwmon(channel, duty)
```

Comment:
This will fail if the hwmon driver is present, but doesn't expose the necessary PWM attributes.

I think we should instead automatically fallback to direct access in that case.